staging/src/k8s.io/apimachinery/pkg/util/yaml/decoder.go

```diff
@@ -327,6 +327,20 @@ func (r *YAMLReader) Read() ([]byte, error) {
 				}
 			}
 			if buffer.Len() != 0 {
+				// If we only have one byte in buffer, check if it is a blank line
```

Comment:
Is this (preexisting) code broken by document separators that appear in block quotations?

Reply:
Does "document separators that appear in block quotations" mean something like this?

```yaml
"---"
stuff: 1

"---"
stuff: 2
```

If so, then these still work. This PR does not change how document separators are detected.

Reply:
Here's an example string, but I'm not sure if it is the most pathological example possible:

```
example: |
  text line one
  text line two
  ---
  text line four
```